doc/FUZZING.md

```diff
@@ -0,0 +1,59 @@
+# Fuzz testing
+
+C-lightning currently supports coverage-guided fuzz testing using [LLVM's libfuzzer](https://www.llvm.org/docs/LibFuzzer.html)
+when built with clang.
+
+
+## Build the fuzz targets
+
+In order to build the C-lightning binaries with code coverage you will need a recent
+[clang](http://clang.llvm.org/). The younger the compiler version the better.
+
+Then you'll need to enable support at configuration time. You likely want to enable
+a few sanitizers for bug detections as well as experimental features for an extended
+coverage (no required though).
+
+```
+DEVELOPER=1 EXPERIMENTAL_FEATURES=1 ASAN=1 UBSAN=1 VALGRIND=0 FUZZING=1 CC=clang ./configure && make
+```
+
+The targets will be built in `tests/fuzz/` as `fuzz-` binaries.
+
+
+## Run one or more target(s)
+
+You can run each target independently. Pass `-help=1` to see available options, for
+example:
+```
+./tests/fuzz/fuzz-addr -help=1
+```
+
+Otherwise, you can use the Python runner to either run the targets against a given seed
+corpus:
+```
+./tests/fuzz/run.py fuzz_corpus -j2
+```
+Or extend this corpus:
+```
+./tests/fuzz/run.py fuzz_corpus -j2 --generate --runs 12345
+```
+
+The latter will run all targets two by two `12345` times.
```

Comment:
i don't understand what `two by two 12345 times` means.

Reply:
So each process does 12345 runs? Maybe something like 

```
Each process runs all targets X times, so `-j2 --runs 12345` will result in 24690 total fuzz attempts.
```

Reply:
If i'm understanding this correctly, this is counter-intuitive to how most parallelization works. I'd expect the total runs to remain constant, with the work divided amongst the different processes (so speeding it up by half instead of doubling the work).

Reply:
Actually the work isn't divided, each process is really independant. The reason to have a multiprocess pool is just that each process can use only one core and be CPU bound (usually RAM usage on fuzz targets is kept low), so you can run many of them in parallel (`X` times each).

Reply:
I understand how it works, thanks for the explainer. My point is that it *shouldn't* work this way -- the total runs should be divided amongst the parallel processes. This is how -n=X works for any other test suite.

Reply:
> I understand how it works, thanks for the explainer.

.. Ok i didn't get what you proposed. I think that does not make sense: you want to be able to run each targets `X` times to generate the most coverage possible. Having a constant total runs across all the targets seems really counter-intuitive: what if you add two new targets ? You need to recompute the ratio to be able to execute each target the wanted number of times ? Same if we add an `--exclude` argument [like bitcoind](https://github.com/bitcoin/bitcoin/blob/9ad7cd2887abf5f91495337a2458560de10b4f69/test/fuzz/test_runner.py#L34-L38) to exclude targets that are too slow to generate coverage, you need to recompute your global value again.
This is really just an exposure of the `-runs` argument of each target.